block-producer/src/state_view/mod.rs

```diff
@@ -44,10 +45,18 @@ where
     S: Store,
 {
     // TODO: Verify proof as well
+    #[allow(clippy::blocks_in_conditions)] // Workaround of `instrument` issue
+    #[instrument(skip_all, err(Debug), fields(COMPONENT = crate::COMPONENT))]
     async fn verify_tx(
```

Comment:
I think for errors we should be using `err(Display)` as this will print out more user-friendly error messages. Or is there a reason to use `Debug` here?

Reply:
@bobbinth at the moment `Display` is not implemented or `VerifyTxError`. We should migrate to `thiserror` and implement pretty printing of error messages, but I'd prefer to do it in separate task.

Reply:
Agreed. I think for now we can add a simple `Display` implementation which just invokes `Debug` printer. And then we can come back and implement `thiserror` later without needing to go back and update tracing code.